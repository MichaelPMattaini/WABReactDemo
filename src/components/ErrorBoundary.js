import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ErrorMessage from './ErrorMessage';

import { sentry } from '../utils';

const propTypes = {
  children: PropTypes.element.isRequired,
};

export default class ErrorBoundary extends Component {
  state = {
    error: null,
  };

  componentDidCatch(error, errorInfo) {
    this.setState({ error });
    sentry.captureException(error, { extra: errorInfo });
  }

  render() {
    if (this.state.error) {
      return <ErrorMessage />;
    }
    return this.props.children;
  }
}

ErrorBoundary.propTypes = propTypes;
