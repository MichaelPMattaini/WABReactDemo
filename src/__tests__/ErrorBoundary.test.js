import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ErrorBoundary from '../components/ErrorBoundary';

configure({ adapter: new Adapter() });

describe('<ErrorBoundary />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <ErrorBoundary>
        <div>App</div>
      </ErrorBoundary>,
      div,
    );
  });

  it('should display children without error', () => {
    const tree = renderer
      .create(
        <ErrorBoundary>
          <div>App</div>
        </ErrorBoundary>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display error message when there is error', () => {
    const component = shallow(
      <ErrorBoundary>
        <div>App</div>
      </ErrorBoundary>,
    );
    component.setState({
      error: true,
    });

    expect(toJson(component)).toMatchSnapshot();
  });
});
