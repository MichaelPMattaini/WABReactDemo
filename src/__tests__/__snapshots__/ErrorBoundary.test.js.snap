// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> should display children without error 1`] = `
<div>
  App
</div>
`;

exports[`<ErrorBoundary /> should display error message when there is error 1`] = `<ErrorMessage />`;
